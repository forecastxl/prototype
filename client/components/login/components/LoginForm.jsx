import React, { PropTypes } from 'react'
import RaisedButton from 'material-ui/RaisedButton'
import TextField from 'material-ui/TextField'
import { ErrorList, TextFieldErrors } from '../../errors'

function LoginForm({ onSubmit, onChange, user, errors }) {
  return (
    <div>
      {errors && errors.base && <ErrorList errors={errors.base} />}
      <form action="/" onSubmit={onSubmit}>
        <TextField
          floatingLabelText="Email"
          name="email"
          value={user.email}
          onChange={onChange}
          errorText={errors && errors.email && <TextFieldErrors errors={errors.email} />}
        /><br />
        <TextField
          floatingLabelText="Password"
          name="password"
          type="password"
          value={user.password}
          onChange={onChange}
          errorText={errors && errors.password && <TextFieldErrors errors={errors.password} />}
        /><br />
        <RaisedButton type="submit" label="Log in" primary />
      </form>
    </div>
  )
}

LoginForm.propTypes = {
  errors: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  user: PropTypes.shape({
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired
  })
}

export default LoginForm
