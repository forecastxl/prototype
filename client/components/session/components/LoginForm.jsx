import React, { PropTypes } from 'react'
import styled from 'styled-components'
import { RaisedButton, TextField } from 'material-ui'

const Aligner = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`

function LoginForm({ onSubmit, onChange, user, errors }) {
  return (
    <Aligner>
      <form action="/" onSubmit={onSubmit}>
        <TextField
          floatingLabelText="Email"
          name="email"
          value={user.email}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        /><br />
        <TextField
          floatingLabelText="Password"
          name="password"
          type="password"
          value={user.password}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        /><br />
        <RaisedButton type="submit" label="Log in" primary />
      </form>
    </Aligner>
  )
}

LoginForm.propTypes = {
  errors: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  user: PropTypes.shape({
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired
  })
}

export default LoginForm
