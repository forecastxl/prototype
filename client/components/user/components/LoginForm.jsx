import React, { PropTypes } from 'react'
import { Link } from 'react-router'
import RaisedButton from 'material-ui/RaisedButton'
import TextField from 'material-ui/TextField'
import { ErrorList, TextFieldErrors } from '../../errors'

function LoginForm({ onSubmit, onChange, user, errors }) {
  return (
    <div>
      {errors && errors.base && <ErrorList errors={errors.base} />}
      <form action="/" onSubmit={onSubmit}>
        <TextField
          floatingLabelText="Email"
          name="email"
          value={user.email}
          onChange={onChange}
          errorText={errors && errors.email && <TextFieldErrors errors={errors.email} />}
        /><br />
        <TextField
          floatingLabelText="Password"
          name="password"
          type="password"
          value={user.password}
          onChange={onChange}
          errorText={errors && errors.password && <TextFieldErrors errors={errors.password} />}
        /><br />
        <RaisedButton type="submit" label="Log in" primary />
      </form>
      <Link to="/login/forgot-password">Forgot your password?</Link>
    </div>
  )
}

LoginForm.propTypes = {
  errors: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  user: PropTypes.shape({
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired
  }).isRequired
}

export default LoginForm
