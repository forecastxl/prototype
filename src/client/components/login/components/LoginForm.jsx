import React, { PropTypes } from 'react'
import { Field, reduxForm } from 'redux-form'
import { onSubmitActions } from 'redux-form-submit-saga'
import { Alert } from '../../alert'
import { Button, TextField } from '../../form'

export function LoginForm({ handleSubmit, pristine, submitting, error }) {
  return (
    <form onSubmit={handleSubmit}>
      {error && <Alert message={error} type={'error'} />}
      <Field name="email" component={TextField} label="Email" fullWidth />
      <Field name="password" component={TextField} label="Wachtwoord" type="password" fullWidth />
      <Button type="submit" label="Inloggen" primary fullWidth disabled={pristine || submitting} />
    </form>
  )
}

LoginForm.propTypes = {
  error: PropTypes.string,
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired
}

LoginForm.defaultProps = {
  error: ''
}

export default reduxForm({
  form: 'login',
  onSubmit: onSubmitActions('LOGIN_SUBMIT', 'LOGIN_SUCCESS', 'LOGIN_VALIDATION_FAILURE')
})(LoginForm)
