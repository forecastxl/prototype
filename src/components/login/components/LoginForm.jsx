import React, { PropTypes } from 'react'
import { TextField, RaisedButton } from 'material-ui'

function LoginForm({ onSubmit, onChange, user }) {
  return (
    <form action="/" onSubmit={onSubmit}>
      <TextField
        floatingLabelText="Email"
        name="email"
        value={user.email}
        onChange={onChange}
      /><br />
      <TextField
        floatingLabelText="Password"
        name="password"
        type="password"
        value={user.password}
        onChange={onChange}
      /><br />
      <RaisedButton type="submit" label="Log in" primary />
    </form>
  )
}

LoginForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  user: PropTypes.shape({
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired
  })
}

export default LoginForm
