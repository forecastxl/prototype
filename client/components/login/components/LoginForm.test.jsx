import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import LoginForm from './LoginForm'

describe('<LoginForm />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <LoginForm
        errors={{}}
        onChange={() => {}}
        onSubmit={() => {}}
        user={{
          email: '',
          password: ''
        }}
      />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
