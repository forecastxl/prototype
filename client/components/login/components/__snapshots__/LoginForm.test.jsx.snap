// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    fullWidth={true}
    label="Email"
    name="email"
  />
  <Field
    component={[Function]}
    fullWidth={true}
    label="Wachtwoord"
    name="password"
    type="password"
  />
  <RaisedButton
    disabled={false}
    fullWidth={true}
    label="Inloggen"
    labelPosition="after"
    primary={true}
    secondary={false}
    type="submit"
  />
</form>
`;
